### PR TITLE
Create api review script changes to add a param to set release tag

### PR DESCRIPTION
**Pull request by praveenkuttappan**
Send a new param in request to APIView to tag a revision as released.

### CODE REVIEW

**Review by weshaggard — APPROVED** [review]

**Comment by praveenkuttappan** [comment]
@weshaggard I have updated the common script to create API review using pregenerated token file as well as source artifact. Once it's merged, I will update python pipeline to use this common script instead of language repo specific script.

File: eng/common/pipelines/templates/steps/create-apireview.yml [hunk 1]
```diff
@@ -2,28 +2,37 @@ parameters:
   ArtifactPath: $(Build.ArtifactStagingDirectory)
   Artifacts: []
   ConfigFileDir: $(Build.ArtifactStagingDirectory)/PackageInfo
+  MarkPackageAsShipped: $false
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/create-apireview.yml:5]:
```suggestion
  MarkPackageAsShipped: false
```

Let's try not to mix the PS-isms in the Devops template.

File: eng/common/pipelines/templates/steps/create-apireview.yml [hunk 2]
```diff
@@ -2,28 +2,37 @@ parameters:
...
           -APIKey $(azuresdk-apiview-apikey)
-          -APILabel "Auto Review - $(Build.SourceVersion)"
-          -PackageName ${{artifact.name}}
+          -PackageName '${{parameters.PackageName}}'
           -SourceBranch $(Build.SourceBranchName)
           -DefaultBranch $(DefaultBranch)
           -ConfigFileDir '${{parameters.ConfigFileDir}}'
+          -BuildId $(Build.BuildId)
+          -RepoName '$(Build.Repository.Name)'    
+          -MarkPackageAsShipped ${{parameters.MarkPackageAsShipped}}
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/create-apireview.yml:32]:
```suggestion
          -MarkPackageAsShipped $${{parameters.MarkPackageAsShipped}}
```

File: eng/common/scripts/Create-APIReview.ps1 [hunk 3]
```diff
@@ -41,6 +51,13 @@ function Submit-APIReview($packagename, $filePath, $uri, $apiKey, $apiLabel, $re
     $versionContent.Headers.ContentDisposition = $versionParam
     $multipartContent.Add($versionContent)
     Write-Host "Request param, packageVersion: $packageVersion"
+    
+    $releaseTagParam = [System.Net.Http.Headers.ContentDispositionHeaderValue]::new("form-data")
+    $releaseTagParam.Name = "setReleaseTag"
```
Comment (weshaggard) [inline eng/common/scripts/Create-APIReview.ps1:56]: I assume this is the name of the API in APIView? Ideally we update it as well but I understand that means an update to the service.

Comment (praveenkuttappan) [reply eng/common/scripts/Create-APIReview.ps1:56]: Correct.

File: eng/common/scripts/Helpers/ApiView-Helpers.ps1 [hunk 4]
```diff
@@ -62,4 +62,51 @@ function Check-ApiReviewStatus($packageName, $packageVersion, $language, $url, $
   {
     Write-Warning "Failed to check API review status for package $($PackageName). You can check http://aka.ms/azsdk/engsys/apireview/faq for more details on API Approval."
   }
+}
+
+function Process-ReviewStatusCode($statusCode, $packageName, $apiApprovalStat, $packageNameStat)
```
Comment (weshaggard) [inline eng/common/scripts/Helpers/ApiView-Helpers.ps1:67]: I'd recommend the full name Status vs Stat on these variables. Stat might lead people to think about statistics instead.

File: eng/common/scripts/Create-APIReview.ps1 [hunk 5]
```diff
@@ -72,119 +89,225 @@ function Submit-APIReview($packagename, $filePath, $uri, $apiKey, $apiLabel, $re
...
 
-$packages = @{}
-if ($FindArtifactForApiReviewFn -and (Test-Path "Function:$FindArtifactForApiReviewFn"))
-{
-    $packages = &$FindArtifactForApiReviewFn $ArtifactPath $PackageName
+    return $StatusCode
 }
-else
+
+function GetAPI-TokenFileName($packageName)
```
Comment (weshaggard) [inline eng/common/scripts/Create-APIReview.ps1:124]:
```suggestion
function Get-APITokenFileName($packageName) 
```

File: eng/common/scripts/Create-APIReview.ps1 [hunk 6]
```diff
@@ -72,119 +89,225 @@ function Submit-APIReview($packagename, $filePath, $uri, $apiKey, $apiLabel, $re
     return $StatusCode
 }
 
+function Uploaded-ReviewTokenFile($packageName, $apiLabel, $releaseStatus, $reviewFileName, $packageVersion)
```
Comment (weshaggard) [inline eng/common/scripts/Create-APIReview.ps1:92]:
```suggestion
function Upload-ReviewTokenFile($packageName, $apiLabel, $releaseStatus, $reviewFileName, $packageVersion)
```

**Pull request merged**